src/src/PluginStructs/P095_data_struct.h

```diff
@@ -38,60 +75,92 @@ enum class ILI9xxx_type_e : uint8_t {
+                   uint32_t            displayTimer,
+                   String              commandTrigger,
+                   uint16_t            fgcolor      = ADAGFX_WHITE,
+                   uint16_t            bgcolor      = ADAGFX_BLACK,
+                   bool                textBackFill = true);
   ~P095_data_struct();
 
-  // Print some text
-  // param [in] string : The text to display
-  // param [in] X : The left position (X)
-  // param [in] Y : The top position (Y)
-  // param [in] textSize : The text size (default 1)
-  // param [in] color : The fore color (default ILI9341_WHITE)
-  // param [in] bkcolor : The background color (default ILI9341_BLACK)
-
-  void printText(const String & string,
-                 int            X,
-                 int            Y,
-                 unsigned int   textSize = 1,
-                 unsigned short color    = ILI9341_WHITE,
-                 unsigned short bkcolor  = ILI9341_BLACK);
-
-  // Parse color string to ILI9341 color
-  // param [in] s : The color string (white, red, ...)
-  // return : color (default ILI9341_WHITE)
-  unsigned short ParseColor(String& s);
-
-  // Split a string by delimiter
-  // param [in] s : The input string
-  // param [in] c : The delimiter
-  // param [out] op : The resulting string array
-  // param [in] limit : The maximum strings to find
-  // return : The string count
-
-  int StringSplit(String& s,
-                  char    c,
-                  String  op[],
-                  int     limit);
+  bool plugin_init(struct EventStruct *event);
+  bool plugin_exit(struct EventStruct *event);
+  bool plugin_read(struct EventStruct *event);
+  bool plugin_write(struct EventStruct *event,
+                    const String      & string);
+  bool plugin_ten_per_second(struct EventStruct *event);
+  bool plugin_once_a_second(struct EventStruct *event);
+
+  void registerButtonState(uint8_t newButtonState,
+                           bool    bPin3Invers);
+  void markButtonStateProcessed();
+  bool getButtonState() {
+    return ButtonState;
+  }
+
+private:
+
+  void displayOnOff(bool state);
+  void updateFontMetrics();
 
   Adafruit_ILI9341 *tft = nullptr;
 
-private:
+  AdafruitGFX_helper *gfxHelper = nullptr;
 
   ILI9xxx_type_e _displayType;
   uint16_t       _xpix;
   uint16_t       _ypix;
+  uint16_t       _textcols;
+  uint16_t       _textrows;
+  uint8_t        _fontwidth    = 6; // Default font characteristics
+  uint8_t        _fontheight   = 10;
+  uint8_t        _heightOffset = 0;
+
+  uint8_t             _rotation;
+  uint8_t             _fontscaling;
+  AdaGFXTextPrintMode _textmode;
+  int8_t              _backlightPin;
+  uint8_t             _backlightPercentage;
+  uint32_t            _displayTimer;
+  uint32_t            _displayTimeout;
+  String              _commandTrigger;
+  uint16_t            _fgcolor;
+  uint16_t            _bgcolor;
+  bool                _textBackFill;
```

Comment:
Shouldn't these be properly initialized to some value?

Reply:
The constructor takes care of that.

Reply:
By initializing in the .h when declaring the variables, you will be sure it is always set, no matter which constructor is being used.
And unless you declare other constructors explicitly with `= delete;` you can't be sure those will not be used by accident. (e.g. assignment constructor)